d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerState.java

```diff
@@ -715,6 +716,15 @@ public void setDelayedExecution(long delayedExecution)
       _weightedSubsetsCache.computeIfAbsent(serviceName, k -> new ConcurrentHashMap<>());
       _weightedSubsetsCache.get(serviceName).put(partitionId, subsetClients);
 
+      debug(_log, "cluster subset updated for service ", serviceName, ": [",
+          subsetClients.values().stream()
+            .limit(LOG_SUBSET_MAX_SIZE)
+            .map(client -> client.getUri() + ":" + client.getSubsetWeight(partitionId))
+            .collect(Collectors.joining(",")),
+          (subsetClients.size() > LOG_SUBSET_MAX_SIZE ? "...(total " + subsetClients.size() + ")" : ""),
```

Comment:
Should we alway print the subset size? Otherwise if it prints fewer than 20 clients, we have to manually count the client in order to understand the subsetting size.

Reply:
Good point. I think always print out the subset size makes sense